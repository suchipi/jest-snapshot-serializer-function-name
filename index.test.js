expect.addSnapshotSerializer(require("./index"));

describe("jest-snapshot-serializer-function-name", () => {
  it("works for anonymous functions", () => {
    expect(() => {}).toMatchSnapshot();
  });

  it("works for named arrow function expressions", () => {
    const namedArrowExpression = () => {};
    expect(namedArrowExpression).toMatchSnapshot();
  });

  it("works for named function declarations", () => {
    function namedDeclaration() {}
    expect(namedDeclaration).toMatchSnapshot();
  });

  it("works for named function expressions", () => {
    const namedExpression = function() {};
    expect(namedExpression).toMatchSnapshot();
  });

  it("works for anonymous functions that were given a name later", () => {
    const renamed = Object.defineProperty(() => {}, "name", {
      value: "new name",
    });
    expect(renamed).toMatchSnapshot();
  });
});
