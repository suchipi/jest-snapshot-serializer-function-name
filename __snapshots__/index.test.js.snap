// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-snapshot-serializer-function-name works for anonymous functions 1`] = `[Function anonymous]`;

exports[`jest-snapshot-serializer-function-name works for anonymous functions that were given a name later 1`] = `[Function new name]`;

exports[`jest-snapshot-serializer-function-name works for named arrow function expressions 1`] = `[Function namedArrowExpression]`;

exports[`jest-snapshot-serializer-function-name works for named function declarations 1`] = `[Function namedDeclaration]`;

exports[`jest-snapshot-serializer-function-name works for named function expressions 1`] = `[Function namedExpression]`;
